File: frontend/lib/models/notification_pref_model.dart
class NotificationPref {
  bool soundEnabled;
  bool vibrationEnabled;
  int snoozeMinutes;
  bool missedDoseAlerts;
  String soundName;

  NotificationPref({
    this.soundEnabled = true,
    this.vibrationEnabled = true,
    this.snoozeMinutes = 10,
    this.missedDoseAlerts = true,
    this.soundName = 'Chime',
  });

  factory NotificationPref.fromJson(Map<String, dynamic> json) {
    return NotificationPref(
      soundEnabled: json['soundEnabled'] ?? true,
      vibrationEnabled: json['vibrationEnabled'] ?? true,
      snoozeMinutes: json['snoozeMinutes'] ?? 10,
      missedDoseAlerts: json['missedDoseAlerts'] ?? true,
      soundName: json['soundName'] ?? 'Chime',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'soundEnabled': soundEnabled,
      'vibrationEnabled': vibrationEnabled,
      'snoozeMinutes': snoozeMinutes,
      'missedDoseAlerts': missedDoseAlerts,
      'soundName': soundName,
    };
  }

  NotificationPref copyWith({
    bool? soundEnabled,
    bool? vibrationEnabled,
    int? snoozeMinutes,
    bool? missedDoseAlerts,
    String? soundName,
  }) {
    return NotificationPref(
      soundEnabled: soundEnabled ?? this.soundEnabled,
      vibrationEnabled: vibrationEnabled ?? this.vibrationEnabled,
      snoozeMinutes: snoozeMinutes ?? this.snoozeMinutes,
      missedDoseAlerts: missedDoseAlerts ?? this.missedDoseAlerts,
      soundName: soundName ?? this.soundName,
    );
  }
}

File: backend/src/models/notification_pref.model.js
const { DataTypes } = require("sequelize");
const { sequelize } = require("../config/db");

const NotificationPref = sequelize.define(
  "NotificationPref",
  {
    id: {
      type: DataTypes.BIGINT.UNSIGNED,
      autoIncrement: true,
      primaryKey: true,
    },
    userId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: false,
      unique: true,
      references: {
        model: "users",
        key: "id",
      },
    },
    soundEnabled: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
    },
    vibrationEnabled: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
    },
    snoozeMinutes: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 10,
      validate: {
        min: { args: [1], msg: "Snooze minutes must be at least 1." },
        max: { args: [60], msg: "Snooze minutes cannot exceed 60." },
      },
    },
    missedDoseAlerts: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: true,
    },
    soundName: {
      type: DataTypes.STRING(50),
      allowNull: false,
      defaultValue: "Chime",
    },
  },
  {
    tableName: "notification_prefs",
    timestamps: true,
    createdAt: "createdAt",
    updatedAt: "updatedAt",
  }
);

module.exports = NotificationPref;